Move "Middleware" to top level in MIQ Navigation

ammendonca opened the pull request:
Moved from being a child of "Compute" to a top-level entry.
Fixes #8110

ammendonca commented:
The PR only moves into top level, but it seems the icon is not appropriate (fa-plus, see screenshot). Should I update it to something else ? Not sure there's a designed icon for MW, if not I've tried some options on the right side, let me know if I should use one of them.

![miq-mw-icons](https://cloud.githubusercontent.com/assets/1737954/15590417/999c8054-238f-11e6-831e-11b1d562bbaf.png)

chessbyte commented:
@jonnyfiveiq thoughts?

abonas commented:
> @jonnyfiveiq thoughts?

@chessbyte it was discussed and approved by @jonnyfiveiq , please see here https://github.com/ManageIQ/manageiq/issues/8110#issuecomment-221848525

@miq-bot add_label providers/hawkular

abonas commented:
@ammendonca indeed the plus icon is not relevant, good catch! we'll need another icon.
As per other suggestions, the cubes are used for container groups (kubernetes pods) so we can't use that for middleware, "tasks" are looking like a server/bare metal icon, and "share" is imo somewhat misleading because it is for share actions :smile: 

@theute @pilhuhn anything standard that can be reused from hawkular/middleware for the icon?
@epwinchell fyi we'll need another icon here.
I'll request some ux feedback on this.

abonas commented:
@ammendonca code wise LGTM.
@martinpovolny please also review given that it touches the main menu

abonas commented:
cc @catrobson do we have a middleware icon that can be used here?

theute commented:
I'm not aware of any "standard" icon we can use here, we'll definitely need UXD team input

abonas commented:
How about the pficon-enterprise icon from [here](https://www.patternfly.org/styles/icons/)?

martinpovolny commented:
You need to:
 * handle remembering of last active menu section/item -- it's not written in a generic way, needs to be updated to match the changes to menu structure
 * handle the default selected menu item (when someone clicks on "Middleware") -- that is also not done in a generic way
* also you need to check the RBAC tree in Role editor -- I tried to modify that to be more generic, but make sure it works

epwinchell commented:
@serenamarie125

serenamarie125 commented:
Middleware should definitely be top level.  I'll take the action item to determine the correct icon for that.

pilhuhn commented:
On 31 May 2016, at 15:53, Serena Chechile Doyle wrote:

> Middleware should definitely be top level.  I'll take the action item 
> to determine the correct icon for that.

Can we use the enterprise icon in the meanwhile and later replace it if 
we find something better?

-- 
hwr@pilhuhn.de  m:0179/207 4919  b:http://pilhuhn.blogspot.com

ammendonca commented:
@martinpovolny, the RBAC tree in Role editor seems to work properly, as per screenshot:

![image](https://cloud.githubusercontent.com/assets/1737954/15677576/bff5cefc-2742-11e6-8d28-b770d3827770.png)

Regarding the first two points, it seems to work fine in case I understood what is required:

- When clicking on "Middleware", we're taken into the "Middleware Providers" page (and the menu is properly selected)
![image](https://cloud.githubusercontent.com/assets/1737954/15677695/4813110a-2743-11e6-8484-e313500c677e.png)
- Reloading the page keeps the selection;
- Navigating to a different MW section (eg: MW Deployments), navigating to a different top menu entry (eg: Networks) and then clicking "Middleware" again, takes it back to the previously selected MW section (Deployments in this case);
- Reloading the page when in this non-default MW section, also keeps the same section and correct menu selection;

Can you please clarify if there's more to test or if I missed the goals of the first two points ?

martinpovolny commented:
so this line is going to be wrong for sure: https://github.com/ManageIQ/manageiq/pull/8792/files#diff-55c5b7aecfb519d0e4880eaf2788eb6eL1943

martinpovolny commented:
and the 2nd: not sure if you want this: https://github.com/ManageIQ/manageiq/blob/master/app/controllers/dashboard_controller.rb#L111

ammendonca commented:
You probably mean this line: https://github.com/ManageIQ/manageiq/pull/8792/files#diff-55c5b7aecfb519d0e4880eaf2788eb6eL1947

Any way I can see/test how this impact the application ?

As for the 2nd, @abonas will we want that ?

abonas commented:
> As for the 2nd, @abonas will we want that ?

what is this code exactly doing? redirecting to the middleware ui?

ammendonca commented:
I am not 100% sure what it does, but I'm guessing the need for it shouldn't be impacted by the move of Middleware to top-level ?

Updated the commit with the removal of affecting the compute tab state with middleware entries (although the impact was likely invisible, as there's no such entry in compute anymore, it was clearly wrong to keep it) and replaced the fa-plus with the suggested pficon-enterprise.

If all is good, should be ready to merge.

martinpovolny commented:
The line means that if you click on middleware submenu, then that click would be remembered as the last click under compute. This code needs to be generalized using the structure defined in `app/presenters/menu/default'.

martinpovolny commented:
The 2nd place sets the default redirect to item when none is remembered. Again this would need to be replaced by some generic code that would cake that place from the default menu definition.

martinpovolny commented:
Otherwise this should be good to merge.

abonas commented:
@ammendonca please address @martinpovolny comments + add an updated screenshot with the enterprise icon.
@serenamarie125 @catrobson I would appreciate your inputs on the [icon suggestion](https://github.com/ManageIQ/manageiq/pull/8998#issuecomment-222512845)

ammendonca commented:
Here's the screenshot with the pficon-enterprise:

![image](https://cloud.githubusercontent.com/assets/1737954/15740688/d652b27a-28ad-11e6-98d9-4af23366e052.png)

I have already fixed remembering the last middleware item associated with compute, with https://github.com/ManageIQ/manageiq/pull/8998/commits/e464a8e8e96bf349f30e40507930cdf396689d53#diff-55c5b7aecfb519d0e4880eaf2788eb6eR1942

As per making any change to https://github.com/ManageIQ/manageiq/blob/master/app/controllers/dashboard_controller.rb#L111, I don't see what's needed, at least in the scope of this issue. This seems to still be applicable, independently of the position of "Middleware" in the menu.

jonnyfiveiq commented:
That works well.

> On 2 Jun 2016, at 10:47, Alexandre Mendonça <notifications@github.com> wrote:
> 
> Here's the screenshot with the pficon-enterprise:
> 
>  <https://cloud.githubusercontent.com/assets/1737954/15740688/d652b27a-28ad-11e6-98d9-4af23366e052.png>
> I have already fixed remembering the last middleware item associated with compute, with e464a8e#diff-55c5b7aecfb519d0e4880eaf2788eb6eR1942 <https://github.com/ManageIQ/manageiq/pull/8998/commits/e464a8e8e96bf349f30e40507930cdf396689d53#diff-55c5b7aecfb519d0e4880eaf2788eb6eR1942>
> As per making any change to https://github.com/ManageIQ/manageiq/blob/master/app/controllers/dashboard_controller.rb#L111 <https://github.com/ManageIQ/manageiq/blob/master/app/controllers/dashboard_controller.rb#L111>, I don't see what's needed, at least in the scope of this issue. This seems to still be applicable, independently of the position of "Middleware" in the menu.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/ManageIQ/manageiq/pull/8998#issuecomment-223245543>, or mute the thread <https://github.com/notifications/unsubscribe/AB_NPbc8o7ZTSMs6tfFWf9WLRrWWRLV8ks5qHqa3gaJpZM4In9fg>.
>

abonas commented:
LGTM

catrobson commented:
Looks good @abonas .  Thanks!

serenamarie125 commented:
@pilhuhn @abonas Good approach to use that icon for now.  As I mentioned earlier, we will put in a formal request for "middleware" to our visuals team and we can replace later on.  Thanks!
